components/suggestion/search_channel_provider.jsx

```diff
@@ -6,12 +6,22 @@ import React from 'react';
 import {autocompleteChannelsForSearch} from 'actions/channel_actions.jsx';
 import AppDispatcher from 'dispatcher/app_dispatcher.jsx';
 import {sortChannelsByDisplayName} from 'utils/channel_utils.jsx';
-import {ActionTypes} from 'utils/constants.jsx';
+import Constants from 'utils/constants.jsx';
 import {localizeMessage} from 'utils/utils.jsx';
 
 import Provider from './provider.jsx';
 import Suggestion from './suggestion.jsx';
 
+function itemToName(item) {
+    if (item.type === Constants.DM_CHANNEL) {
+        return '@' + item.display_name;
```

Comment:
Can you confirm for me that `display_name` is always the `username`, and not sometimes rendered as the `FirstName LastName`? (Same for group below.)

Reply:
the display name of a direct channel, normally is empty, but In this case I generate the username there for the autocompletion. In the groups, the display_name is the concatenated list of members of the group, which is used to the autocompletion. It is for autocompletion so makes sense to show what you will autocomplete.